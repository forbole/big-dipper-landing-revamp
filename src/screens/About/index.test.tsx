/* eslint-disable react/display-name */
import Box from '@mui/material/Box';
import { render, screen } from '@testing-library/react';
import { MockTheme } from '~tests/mocks';
import Component from '.';

const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('~src/components/Layout', () => (props: object) => (
  <Box data-testid="Layout" {...props} />
));
jest.mock('~src/components/SectionBox', () =>
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  ({ main, ...props }: { [p: string]: unknown }) => (
    <Box data-testid="SectionBox" {...props} />
  )
);
jest.mock('~src/components/SectionLimit', () => (props: object) => (
  <Box data-testid="SectionLimit" {...props} />
));
jest.mock('~src/components/ContentBox', () => (props: object) => (
  <Box data-testid="ContentBox" {...props} />
));
// ==================================
// unit tests
// ==================================
describe('About', () => {
  it('matches snapshot', async () => {
    const { container } = render(
      <MockTheme>
        <Component />
      </MockTheme>
    );

    expect(screen.getByText(/about/i)).toBeInTheDocument();
    expect(screen.getByText(/description/i)).toBeInTheDocument();
    expect(screen.getByTestId('Layout')).toBeInTheDocument();
    expect(screen.getByTestId('SectionBox')).toBeInTheDocument();
    expect(screen.getByTestId('SectionLimit')).toBeInTheDocument();
    expect(screen.getByTestId('ContentBox')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
