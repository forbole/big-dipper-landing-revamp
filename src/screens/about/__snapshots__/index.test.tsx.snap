// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About matches snapshot 1`] = `
.emotion-0 .title {
  margin-bottom: 32px;
  text-align: center;
}

@media (min-width:1280px) {
  .emotion-0 .title {
    text-align: left;
    margin-bottom: 40px;
  }
}

.emotion-1 {
  margin: 0;
  font-size: 2rem;
  font-weight: 600;
  font-family: SFPro,Arial,sans-serif;
  line-height: 1.2;
}

.emotion-2 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  font-family: SFPro,Arial,sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

<div>
  <div
    data-testid="Layout"
  >
    <div
      data-testid="SectionBox"
    >
      <div
        class="emotion-0"
        data-testid="SectionLimit"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 title emotion-1"
        >
          about
        </h2>
        <div
          data-testid="ContentBox"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-2"
          >
            description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
